browser/src/canvas/sections/ChartContextButtonSection.ts

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright the Collabora Online contributors.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+class ChartContextButtonSection extends HTMLObjectSection {
+	static readonly namePrefix: string = 'ChartContextButton_';
+	static readonly classNames: string[] = [
+		'chart-select-theme',
+		'chart-save-to-new-theme',
+	];
+
+	constructor(buttonType: number) {
+		super(
+			ChartContextButtonSection.namePrefix + buttonType,
+			32,
+			32,
+			new cool.SimplePoint(
+				GraphicSelection.rectangle.x2 + 5 * 16,
+				GraphicSelection.rectangle.y1 + buttonType * (32 + 5) * 16,
+			),
+			ChartContextButtonSection.classNames[buttonType],
+			true,
+		);
+		this.updatePosition();
+		this.sectionProperties.buttonType = buttonType;
+		this.sectionProperties.lastInputEvent = null;
+	}
+
+	public onMouseEnter(point: cool.SimplePoint, e: MouseEvent): void {
+		this.getHTMLObject()?.classList.add('hovered');
+	}
+
+	public onMouseLeave(point: cool.SimplePoint, e: MouseEvent): void {
+		this.getHTMLObject()?.classList.remove('hovered');
+	}
+
+	public onMouseDown(point: cool.SimplePoint, e: MouseEvent): void {
+		this.stopEvents(e);
+	}
+
+	public onMouseUp(point: cool.SimplePoint, e: MouseEvent): void {
+		this.stopEvents(e);
+	}
+
+	private stopEvents(e: MouseEvent) {
+		this.stopPropagating();
+
+		// We shouldn't need below 2 when we remove map element.
+		e.preventDefault();
+		e.stopImmediatePropagation();
+	}
+
+	public onClick(point: cool.SimplePoint, e: MouseEvent): void {
+		if (this.sectionProperties.buttonType === 0) {
+			app.socket.sendMessage('uno .uno:SelectTheme');
+		} else {
+			app.socket.sendMessage('uno .uno:ChartSaveToNewTheme');
+		}
+	}
+
+	showChartContextToolbar(): void {
+		this.setShowSection(true);
+	}
+
+	updatePosition(): void {
+		var origZoom = app.map.getZoom();
+		var zoomValue = app.map.zoomToFactor(origZoom);
+
+		this.position[0] = Math.round(
```

Comment:
Can you please try something like:
`this.setPosition(x, y);`
Instead of `this.position[0] = .. and this.position[1] = ..`

Also, `myTopLeft` variable is set by setPosition and canvas container. I think you shouldn't need that after using `this.setPosition`.